src/com/puppetlabs/mq.clj

```diff
@@ -132,10 +132,16 @@
     (start-broker! (build-embedded-broker brokername dir config))
     (catch java.io.EOFException e
       (log/warn
-        (str "Caught EOFException on broker startup, trying to restart it "
-             "again to see if that solves it. This is probably due to "
-             "KahaDB corruption."))
-      (start-broker! (build-embedded-broker brokername dir config)))))
+       "Caught EOFException on broker startup, trying again."
+       "This is probably due to KahaDB corruption "
+       "(see \"KahaDB Corruption\" in the PuppetDB manual).")
+      (start-broker! (build-embedded-broker brokername dir config)))
+    (catch java.io.IOException e
+      (log/error
+       e "Unable to start broker in " (pr-str dir) "."
+       "This is probably due to KahaDB corruption or version downgrade "
```

Comment:
Hmm, its not really 'corrupted' per se, indicating a bug or error of some kind, it's just the inability for old software to read the new 5.11.1 KahaDB disk format. I guess corruption makes it sound more fatal than it is. Corruption would be something I'd want to dig into as a user to understand why, if it's just a downgrade incompatibility, hopefully most users wouldn't bother to investigate or raise a ticket on it - the message is less scary.

Honestly, KahaDB is just terrible, I hate making these work-arounds. I hope they sort this kind of thing out themselves one day. LevelDB is another option I've been keeping an eye on, last time I tried it though it was in a worse state as far as dealing with weird scenarios like this, we should look at it again soon though, it's been about a year - hopefully its more stable.